src/UCTSearch.cpp

```diff
@@ -39,9 +39,10 @@
 
 using namespace Utils;
 
-UCTSearch::UCTSearch(GameState & g)
+UCTSearch::UCTSearch(const GameState & g)
     : m_rootstate(g) {
     set_playout_limit(cfg_max_playouts);
+    m_rootstate.disable_history();
```

Comment:
This is the majority of the change, in the MCTS we don't need history.

We also save by not carrying around FullBoard which is much larger than the pair<bitset<361>,bitset<361>> that it get's converted to. 
Additionally we cache 14/18 of the input planes.

What's not yet perfect in this code is the fact that m_rootstate(g) full copies the history then immediately discards it :/